WooCommerce/Classes/ViewModels/Order Details/Shipping Labels/BoldableTextView.swift

```diff
@@ -0,0 +1,113 @@
+import SwiftUI
+
+/// A text view that takes in a string whose substrings within two asterisks are bolded.
+///
+/// For example, initializing with a string "*Never* will I *stop learning*." will bold the substrings "Never" and "stop learning".
+///
+struct BoldableTextView: View {
+    fileprivate struct Element {
+        let content: String
+        let isBold: Bool
+
+        init(content: String, isBold: Bool) {
+            if isBold {
+                self.content = content.replacingOccurrences(of: boldRegionMark, with: "")
+            } else {
+                self.content = content
+            }
+            self.isBold = isBold
+        }
+    }
+
+    fileprivate static let boldRegionMark = "*"
+
+    private let elements: [Element]
+
+    init(_ content: String) {
+        elements = content.parseBoldableElements()
+    }
+
+    var body: some View {
+        elements.map { $0.toTextView }
+            .reduce(into: Text(""), { result, text in
+                result = result + text
+            })
+    }
+}
+
+private extension BoldableTextView.Element {
+    var toTextView: Text {
+        if isBold {
+            return Text(content).fontWeight(.bold)
+        } else {
+            return Text(content)
+        }
+    }
+}
+
+private extension String {
+    /// Parses the input text and returns a collection of boldable elements.
+    ///
+    /// - Returns: A collection of boldable elements.
+    func parseBoldableElements() -> [BoldableTextView.Element] {
+        let boldRegionMark = BoldableTextView.boldRegionMark
+        let regex = try! NSRegularExpression(pattern: "\\\(boldRegionMark){1}(.*?)\\\(boldRegionMark){1}")
+        let range = NSRange(location: 0, length: count)
+
+        /// Finds all the ranges that match the regex for content within a bold region.
+        let matches: [NSTextCheckingResult] = regex.matches(in: self, options: [], range: range)
+        let matchingRanges = matches.compactMap { Range<Int>($0.range) }
```

Comment:
`Range` has a constructor that can convert from `NSRange` to the native range of `String`

```
Range(range: NSRange, in: String)
```

I believe if we operate with this range we don't need a `subscript(range: Range<Int>) -> String` method and can use `substring` method of a string.

Reply:
It looks like `substring` is deprecated in Swift 5, and the range has to be in the index form like what the `String.subscript` helper currently does (also suggested in [this SO response](https://stackoverflow.com/a/46634511/9185596)) - do you have other suggestions on replacing the `String` extension `subscript(range: Range<Int>) -> String`?